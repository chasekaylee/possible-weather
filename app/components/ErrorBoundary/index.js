import React from 'react';
import PropTypes from 'prop-types';

export default class ErrorBoundary extends React.Component {
  static propTypes = {
    children: PropTypes.node,
  };

  constructor(props) {
    super(props);
    this.state = { hasError: false };
  }

  static getDerivedStateFromError() {
    // Update state so the next render will show the fallback UI.
    return { hasError: true };
  }

  componentDidCatch() {
    // You can also log the error to an error reporting service
  }

  render() {
    if (this.state.hasError) {
      // You can render any custom fallback UI
      return <h1>Something went wrong.</h1>;
    }

    return this.props.children;
  }
}
