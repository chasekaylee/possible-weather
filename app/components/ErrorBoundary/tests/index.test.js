import React from 'react';
import { render } from 'react-testing-library';
// import 'jest-dom/extend-expect'; // add some helpful assertions

import ErrorBoundary from '../index';

describe('<ErrorBoundary />', () => {
  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    render(
      <ErrorBoundary>
        <div />
      </ErrorBoundary>,
    );
    expect(spy).not.toHaveBeenCalled();
  });

  it('Should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(
      <ErrorBoundary>
        <div />
      </ErrorBoundary>,
    );
    expect(firstChild).toMatchSnapshot();
  });
});
