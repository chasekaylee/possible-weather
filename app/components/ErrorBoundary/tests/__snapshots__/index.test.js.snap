// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ErrorBoundary /> Should render and match the snapshot 1`] = `<div />`;
